lnwire/dyn_propose_test.go

```diff
@@ -0,0 +1,95 @@
+package lnwire
+
+import (
+	"bytes"
+	"math"
+	"math/rand"
+	"reflect"
+	"testing"
+	"testing/quick"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+)
+
+func (dp DynPropose) Generate(rand *rand.Rand, size int) reflect.Value {
+	var id ChannelID
+	dp.ChanID = id
+	rand.Read(dp.ChanID[:])
+
+	// This coin variable is a coin toss that increases the probability of
+	// nil rows on DynPropose as the size parameter gets smaller
+	var coin int
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := btcutil.Amount(rand.Uint32())
+		dp.DustLimit = &v
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := MilliSatoshi(rand.Uint32())
+		dp.MaxValueInFlight = &v
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := btcutil.Amount(rand.Uint32())
+		dp.ChannelReserve = &v
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := MilliSatoshi(rand.Uint32())
+		dp.HtlcMinimum = &v
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := uint16(rand.Uint32())
+		dp.CsvDelay = &v
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := uint16(rand.Uint32())
+		dp.MaxAcceptedHTLCs = &v
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v, _ := btcec.NewPrivateKey()
+		dp.FundingKey = v.PubKey()
+	}
+
+	coin = rand.Intn(math.MaxInt)
+	if coin < size {
+		v := ChannelType(*NewRawFeatureVector())
+		dp.ChannelType = &v
+	}
+
+	return reflect.ValueOf(dp)
+}
+
+func TestDynProposeRoundTripSerialization(t *testing.T) {
```

Comment:
Any reason not to use the existing quick check tests here? https://github.com/lightningnetwork/lnd/blob/master/lnwire/lnwire_test.go#L353-L357

Reply:
Didn't know they existed when I wrote this. I can adjust